File: 0078-subsets/0078-subsets.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    void solve(int[]nums, int index, List<Integer>temp) {
        if(index >= nums.length) {
            ans.add(new ArrayList<>(temp));
            return;
        }
        ans.add(new ArrayList<>(temp));
        for(int i=index; i<nums.length; i++) {
            temp.add(nums[i]);
            solve(nums, i+1, temp);
            temp.remove(temp.size()-1);
        }
    }
    public List<List<Integer>> subsets(int[] nums) {
        solve(nums, 0, new ArrayList<>());
        return ans;
    }
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>>ans;
    void solve(vector<int>&nums, int ind, vector<int>&temp) {
        if(ind >= nums.size()) return;
        temp.push_back(nums[ind]);
        ans.push_back(temp);
        solve(nums, ind+1, temp);
        temp.pop_back();
        solve(nums, ind+1, temp);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int>temp;
        ans.push_back(temp);
        solve(nums, 0, temp);
        return ans;
    }
};